src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

```diff
@@ -3,10 +3,18 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="Microsoft.Managed.Core.targets"/>
 
-  <PropertyGroup Condition="('$(TargetFrameworkIdentifier)' != '.NETCoreApp' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') AND
-                            ('$(TargetFrameworkIdentifier)' != '.NETStandard' OR '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')">
-    <MaxSupportedLangVersion Condition="'$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
-    <LangVersion Condition="'$(LangVersion)' == ''">$(MaxSupportedLangVersion)</LangVersion>
+  <PropertyGroup>
+    <!-- .NETCoreApp < 3.0, .NETStandard < 2.1, or any other target framework -->
+    <MaxSupportedLangVersion Condition="(('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' &lt; '3.0') OR
+                                        ('$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(_TargetFrameworkVersionWithoutV)' &lt; '2.1')) AND
+                                        '$(MaxSupportedLangVersion)' == ''">7.3</MaxSupportedLangVersion>
+    
+    <!-- .NETCoreApp < 5.0, .NETStandard == 2.1 -->
+    <MaxSupportedLangVersion Condition="(('$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' &lt; '5.0') OR
+                                        ('$(TargetFrameworkIdentifier)' == '.NETStandard' AND '$(_TargetFrameworkVersionWithoutV)' == '2.1')) AND
+                                        '$(MaxSupportedLangVersion)' == ''">8.0</MaxSupportedLangVersion>
+    
+    <LangVersion Condition="'$(LangVersion)' == '' AND '$(MaxSupportedLangVersion)' != ''">$(MaxSupportedLangVersion)</LangVersion>
```

Comment:
What happens when the target framework is .NET Framework? In that case I believe that `$(MaxSupportedLangVersion)` will be empty. 

Think the original code here had the right idea. If there is no `$(MaxSupportedLangVersion)` calculated, that is if it isn't a target framework that we could positively identify, then fall back to 7.3 as the default.

Reply:
It'll be 7.3. We check if its either a) netcore < 3, b) netstandard < 2.1 or it's empty (all other frameworks)

Reply:
Why did you opt to change the original logic here?  For me at least this is much harder to follow.